plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -2048,11 +2046,7 @@ def get_marketplace_image_version(self):
                                                                       self.image['sku'],
                                                                       str(exc)))
         if versions and len(versions) > 0:
-            if self.image['version'] == 'latest':
-                return versions[len(versions) - 1]
-            for version in versions:
-                if version.name == self.image['version']:
-                    return version
+            return versions[len(versions) - 1].name
```

Comment:
If version is specified and cannot be obtained through the first one, the PR function is similar to #767. Thank you very much!

Reply:
With my changes the version lookup will only happen when "latest" is used. Otherwise the user specified version is used directly. I think it doesn't make much sense to get all versions of an image to check if the specified version is listed or not.